packages/terra-pills/src/Pill.jsx

```diff
@@ -0,0 +1,171 @@
+    contentHeight: PropTypes.oneOf(Object.keys(Popup.Opts.heights)),
+    /**
+     * A string representation of the width in px, limited to:
+     * 160, 240, 320, 640, 960, 1280, 1760 or auto.
+     */
+    contentWidth: PropTypes.oneOf(Object.keys(Popup.Opts.widths)),
+    /**
+     * Set this to `true` if your content has focusable elements and you want them to receive focus instead of focusing on the default popup frame when the popup is opened.
+     */
+    isContentFocusDisabled: PropTypes.bool,
+  }),
+};
+
+const defaultProps = {
+  onRemove: undefined,
+  popupConfig: undefined,
+};
+
+const Pill = (props) => {
+  const {
+    labelText,
+    onRemove,
+    popupConfig,
+  } = props;
+
+  const [open, setOpen] = useState(false);
+  const pillRef = useRef();
+
+  const handleOnClick = () => {
+    setOpen(true);
+  };
+
+  const handleOnRemove = () => {
+    pillRef.current.style.display = 'none';
+    onRemove();
+  };
+
+  const handleOnKeyDown = (event) => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'true');
+    if (event.keyCode === KEY_RETURN || event.keyCode === KEY_SPACE) {
+      setOpen(true);
+    } else if ((event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && onRemove) {
+      onRemove();
+    }
+  };
+
+  const handleOnMouseDown = () => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'false');
+  };
+
+  const handleOnRequestClose = () => {
+    setOpen(false);
+  };
+
+  const getPillNode = () => pillRef.current;
+
+  const pillProps = {};
+  if (popupConfig || onRemove) {
+    pillProps.tabIndex = '0';
+    pillProps.onKeyDown = handleOnKeyDown;
+    pillProps.onMouseDown = handleOnMouseDown;
+  }
+
+  const popupButtonProps = {};
+  if (popupConfig) {
+    popupButtonProps.onClick = handleOnClick;
+    popupButtonProps.role = 'button';
+  }
+
+  const removeButtonProps = {};
+  if (onRemove) {
+    removeButtonProps.onClick = handleOnRemove;
+    removeButtonProps.role = 'button';
+  }
+
+  const theme = React.useContext(ThemeContext);
+  const pillClassNames = cx([
+    'pill-container',
+    { 'is-focusable': !!popupConfig || !!onRemove },
+    theme.className,
+  ]);
+  const pillLabelClassNames = cx([
+    'pill-label',
+    { 'is-popup-button': !!popupConfig },
+    { 'is-removable': !!onRemove },
+  ]);
+  const removeButtonClassNames = cx([
+    'remove-button',
+    { 'is-removable': !!onRemove },
+  ]);
+
+  return (
+    <>
+      <div
+        {...pillProps}
+        className={pillClassNames}
+        ref={pillRef}
+      >
+        <div
+          {...popupButtonProps}
+          className={pillLabelClassNames}
+        >
+          {labelText}
+        </div>
+        {onRemove && (
+          <div
+            {...removeButtonProps}
+            className={removeButtonClassNames}
+          >
+            <IconClear height="0.57143rem" width="0.57143rem" />
+          </div>
+        )}
+      </div>
+      {popupConfig && (
```

Comment:
I'd recommend not bundling the Popup with the Pill and simply exposing a ref to the Pill so that consumers can_render a Popup themselves if they want to. This decouples the Popup's inherent complexity from the Pill and gives the Pill less responsibility (and thus less surface area for future breaking changes).

I'd boil the props down to:
`label` -> that's what we're using throughout terra-application for props that denote user-visible text
`onSelect` -> Generic selection callback. If not provided, the Pill is not selectable.
`onSelectRemove` -> Remove button selection callback. If not provided, the Pill is not removable.
`refCallback` -> Callback to expose Pill container element for Popup placement.

Reply:
We talked about this during scrum this morning. @neilpfeiffer and @jeremyfuksa agreed that removing the integrated Popup within the Pill makes sense. However, we still want to be able to create documentation examples that exhibit the desired Popup integration.

Reply:
with the dependency on pop up removed should we move this component back to terra-core?

Reply:
It seems like it becomes a lot more foundational without that dependency. Core feels like the more logical place for this.